### PR TITLE
Excel upgrade mechanism

### PR DESCRIPTION
**Pull request by A-Scott-Rowe**
## Tracker Ticket

* https://quandl.atlassian.net/browse/AP-2339

## Description

* Add an upgrade mechanism based on github releases

## Note to the reviewer

* None

## Note to QA

* None

## Dependency

* None

### CODE REVIEW

File: QuandlExcelAddinWindows/UI/ChangeLog.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    /// <summary>
+    /// Interaction logic for UserControl1.xaml
+    /// </summary>
+    public partial class ChangeLog : UserControl
+    {
+        public ChangeLog(string message)
+        {
+            InitializeComponent();
+
+            var md = new Markdown.Xaml.Markdown();
```
Comment (A-Scott-Rowe) [inline QuandlExcelAddinWindows/UI/ChangeLog.xaml.cs]: We will need to provide this dll with the repo

File: QuandlExcelAddinWindows/UI/Update.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        public Update(Shared.Helpers.Updater updater)
+        {
+            var md = new Markdown.Xaml.Markdown();
+            
+            _updater = updater;
+            
+            InitializeComponent();
+
+
+            DocFlow.Document = md.Transform(_updater.latestRelease.Body);
```
Comment (junos) [inline QuandlExcelAddinWindows/UI/Update.xaml.cs]: there are too many space lines here and above.

File: QuandlExcelAddinWindows/Toolbar.cs [hunk 3]
```diff
@@ -24,7 +33,14 @@ private void Ribbon2_Load(object sender, RibbonUIEventArgs e)
 
         private void AboutButton_Click(object sender, RibbonControlEventArgs e)
         {
-            new AboutForm().Show();
+            if (_updater.UpdateAvailable)
+            {
+
```
Comment (junos) [inline QuandlExcelAddinWindows/Toolbar.cs]: if do nothing here, please adjust the logic and do't use else statement

File: QuandlExcelAddinWindows/Settings.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿namespace Quandl.Excel.Addin.Properties {
+
```
Comment (junos) [inline QuandlExcelAddinWindows/Settings.cs]: remove this file if we don't provide the update option selection

File: Quandl.Excel.Addin.Setup/Quandl.Excel.Addin.Setup.isl [hunk 5]
```diff
@@ -2436,7 +2436,7 @@
...
 		<row><td>DVD-9</td><td>Express</td><td>&lt;ISProjectDataFolder&gt;</td><td>Default</td><td>3</td><td>1033</td><td>0</td><td>2</td><td>Intel</td><td/><td>1033</td><td>0</td><td>7.95</td><td>1</td><td>2048</td><td/><td>0</td><td/><td>MediaLocation</td><td/><td>http://</td><td/><td/><td/><td/><td>75805</td><td/><td/><td/><td>3</td></row>
-		<row><td>SingleImage</td><td>Express</td><td>&lt;ISProjectDataFolder&gt;</td><td>PackageName</td><td>1</td><td>1033</td><td>0</td><td>1</td><td>Intel</td><td/><td>1033</td><td>0</td><td>0</td><td>0</td><td>0</td><td/><td>0</td><td/><td>MediaLocation</td><td/><td>http://</td><td/><td/><td/><td/><td>536979487</td><td/><td/><td/><td>3</td></row>
+		<row><td>SingleImage</td><td>Express</td><td>&lt;ISProjectDataFolder&gt;</td><td>PackageName</td><td>1</td><td>1033</td><td>0</td><td>1</td><td>Intel</td><td/><td>1033</td><td>0</td><td>0</td><td>0</td><td>0</td><td/><td>0</td><td/><td>MediaLocation</td><td/><td>http://</td><td/><td/><td/><td/><td>108573</td><td/><td/><td/><td>3</td></row>
```
Comment (A-Scott-Rowe) [inline Quandl.Excel.Addin.Setup/Quandl.Excel.Addin.Setup.isl]: This just sets the default behaviour for 'sign release package' to none

File: Quandl.Shared.Modules/Utilities.cs [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
 using Newtonsoft.Json.Linq;
+using MoreLinq;
+using System.Drawing;
 
 namespace Quandl.Shared
 {
     public class Utilities
     {
         public const string ReleaseVersion = "Beta 1.0 v3.50";
+        public const int GithubReleaseId = 4655920;
```
Comment (junos) [inline Quandl.Shared.Modules/Utilities.cs]: please update this to a proper value, the current value is: 4691412

**Comment by junos** [comment]
code looks good, GithubReleaseId = 4691412 may still have issue, but let's see 👍

**Pull request merged**